### ISSUE TITLE
-Wmissing-braces in drivers/android/binderfs.c

### ISSUE THREAD

**Issue by nickdesaulniers** [ClangBuiltLinux/linux#674]
Reported-by: kernelci.org bot <bot@kernelci.org>
```
    1    drivers/android/binderfs.c:657:41: warning: suggest braces around initialization of subobject [-Wmissing-braces]
```
cc @ajs1984

**Comment by nathanchance** [comment]
For the record, I do not see this with `clang-9` or `clang-10`, only `clang-8` and earlier. Should it still be silenced with `memset`?

**Comment by nickdesaulniers** [comment]
oh, that is interesting.  KernelCI is reporting a few of these -Wmissing-braces.  I wonder what changed in Clang?

**Comment by nathanchance** [comment]
Looks like https://github.com/llvm/llvm-project/commit/fa52e00c85ce3feeec14be34265781f721b966c0 according to my bisect.

```
$ git bisect log
git bisect start '--term-old' 'broken' '--term-new' 'fixed'
# broken: [7b5565418f4d6e113ba805dad40d471d23bca6f6] Fix build breakage from llvm r351317
git bisect broken 7b5565418f4d6e113ba805dad40d471d23bca6f6
# fixed: [c89a3d78f43d81b9cff7b9248772ddf14d21b749] [lldb][NFC] Format 'type' commands in Options.td
git bisect fixed c89a3d78f43d81b9cff7b9248772ddf14d21b749
# broken: [12a561fa1b79c449d518100e1dd0dfce0a37b65d] [x86] use psubus for more vsetcc lowering (PR39859)
git bisect broken 12a561fa1b79c449d518100e1dd0dfce0a37b65d
# broken: [0338b88861d0a5448518a79b30062b061caee516] [AIX] Implement call lowering with parameters could pass onto GPRs
git bisect broken 0338b88861d0a5448518a79b30062b061caee516
# broken: [6b3f56b65fa1e5094201fedb8a9cb9cd30eac9e5] [WebAssembly] Fix p2align in assembler.
git bisect broken 6b3f56b65fa1e5094201fedb8a9cb9cd30eac9e5
# broken: [dbb6d01d340f5fecd0591f2ef25acb043bd6957b] [MIPS GlobalISel] Regbanks for G_SELECT. Select i64, f32 and f64 select
git bisect broken dbb6d01d340f5fecd0591f2ef25acb043bd6957b
# broken: [51a05d72ae2969b290d1565c9af1b8f6995ed9ea] AMDGPU: Drop remnants of byval support for shaders
git bisect broken 51a05d72ae2969b290d1565c9af1b8f6995ed9ea
# fixed: [74c350af2181cf21085f69bd2d9bcf4d6d2dc03c] [DWARF] Fix an incorrect format specifier.
git bisect fixed 74c350af2181cf21085f69bd2d9bcf4d6d2dc03c
# broken: [8d879c8d95496c0a9049ff80261fbc5754ce56b3] [AMDGPU][MC] Corrected encoding of src0 for DS_GWS_* instructions
git bisect broken 8d879c8d95496c0a9049ff80261fbc5754ce56b3
# broken: [794346460afa212690d0f006023a0e770884f0bb] [clang] allow -fthinlto-index= without -x ir
git bisect broken 794346460afa212690d0f006023a0e770884f0bb
# broken: [fdcbd5fa48680a1f02809d2ead6259b30b00d0b1] Temporarily Revert "fix unnamed fiefield issue and add tests for __builtin_preserve_access_index intrinsic"
git bisect broken fdcbd5fa48680a1f02809d2ead6259b30b00d0b1
# fixed: [a17b1aed6ab205515adc31d19e953635e563e5c4] Add contains method to associative containers. This patch implements P0458R2, adding contains to map, multimap, unordered_map, unordered_multimap, set, multiset, unordered_set, and unordered_multiset.
git bisect fixed a17b1aed6ab205515adc31d19e953635e563e5c4
# broken: [543ba4e9e0c421bedaea2d8a0f1965092cec300e] [InstructionSimplify] Apply sext/trunc after pointer stripping
git bisect broken 543ba4e9e0c421bedaea2d8a0f1965092cec300e
# fixed: [492ce8cc8b0831924f7f0479f37f2e708a8dad3f] reland "add -fthinlto-index= option to clang-cl"
git bisect fixed 492ce8cc8b0831924f7f0479f37f2e708a8dad3f
# fixed: [ca16d280f7ed9509398ba8869a123c9f8ede72f7] Re-land "[DebugInfo] Move function from line table to the prologue (NFC)"
git bisect fixed ca16d280f7ed9509398ba8869a123c9f8ede72f7
# fixed: [fa52e00c85ce3feeec14be34265781f721b966c0] [Sema] Suppress additional warnings for C's zero initializer
git bisect fixed fa52e00c85ce3feeec14be34265781f721b966c0
# first fixed commit: [fa52e00c85ce3feeec14be34265781f721b966c0] [Sema] Suppress additional warnings for C's zero initializer
```

**Comment by nickdesaulniers** [comment]
Hmmm...maintainer of Clang participated in code review of that and the patch it references, so without thinking too much more about it, I assume it's not a regression in clang.  Just something we have to live with pre-clang-9.
